chia/util/network.py

```diff
@@ -20,6 +21,14 @@
 from chia.util.task_referencer import create_referenced_task
 
 
+def parse_host_port(host_port: str) -> tuple[str, int]:
+    """Parse a host:port string into a tuple of (host, port), raising ValueError on failure."""
+    result = urlsplit(f"//{host_port}")
```

Comment:
so, `host_port` is allowed to have more URL components here, and we'll just ignore them?

Reply:
Yeah, it's a little sketch I suppose, but it seems harmless. And it should play better with ipv6.